zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupConfig.java

```diff
@@ -18,28 +18,100 @@
 
 package org.apache.zookeeper.server.backup;
 
+import java.io.File;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.zookeeper.common.ConfigException;
+import org.apache.zookeeper.server.util.VerifyingFileFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 /**
- * Backup-related configurations
+ * Backup-related configurations.
  */
 public class BackupConfig {
-  public static final String NFS_BACKUP_STORAGE = "NFSBackupStorage";
+  private static final Logger LOG = LoggerFactory.getLogger(BackupConfig.class);
+
+  /*
+   * Default constants
+   */
+  private static final boolean DEFAULT_ENABLED = false;
+  public static final int DEFAULT_RETENTION_DAYS = 20;
+  public static final long DEFAULT_BACKUP_INTERVAL_MS = 30 * 60 * 1000L; // 30 minutes
+  /*
+   * For retention maintenance, -1 indicates no maintenance by default.
+   * Some storage backends support TTL natively.
+   */
+  public static final long DEFAULT_RETENTION_MAINTENANCE_INTERVAL_MS = -1L;
```

Comment:
Shall we leave the constants to a constant class?

Reply:
I think the way I approach whether a constant class is needed is this: Are the constants wider in scope and are used in various places throughout the codebase?

If yes, then I think we can justify having a constant class. If not, I think it's cleaner to keep them in the classes they're only used in.

In this case, these values are only confined to backup logic and are not used in the rest of the zookeeper codebase. Additionally, the scope is even narrower within backup since it's only being used in BackupConfig and its corresponding test. So I'd like to keep these constants in this class.